snpdb/signals/variant_search.py

```diff
@@ -179,20 +179,20 @@ def get_results_from_variant_tuples(qs: QuerySet, data: VariantCoordinate, any_a
     :param any_alt: If true, search without using alt and return all matches
     :return: A QuerySet of variants
     """
-    (chrom, position, ref, alt) = data
+    chrom, position, end, ref, alt = data
     position = int(position)
 
-    results = qs.filter(Variant.get_chrom_q(chrom), locus__position=position, locus__ref__seq=ref)
+    results = qs.filter(Variant.get_chrom_q(chrom), locus__position=position, locus__ref__seq=ref, end=end)
     if not any_alt:
         results = results.filter(alt__seq=alt)
 
     if not results:
         if not any_alt:
-            return ModifiedImportedVariant.get_variants_for_unnormalized_variant(chrom, position, ref, alt)
+            return ModifiedImportedVariant.get_variants_for_unnormalized_variant(chrom, position, end, ref, alt)
         else:
             # should we really be searching ModifiedImportVariants with any alt? or should that just happen for
             # the filter of "real" variants
-            return ModifiedImportedVariant.get_variants_for_unnormalized_variant_any_alt(chrom, position, ref)
+            return ModifiedImportedVariant.get_variants_for_unnormalized_variant_any_alt(chrom, position, end, ref)
```

Comment:
Just in general there's a lot of constructing VariantCoordinate only to deconstruct them into chrom, position, end, ref, alt again.

Doesn't need to be fixed this merge - but if ther's good unit test coverage I would like to see more and more methods simply take VariantCoordinate

Reply:
I have converted as many variant tuples to variant coordinate as I could